### ISSUE TITLE
Signed_Transaction cannot handle appbase transactions with beneficiaries

### ISSUE THREAD

**Issue by crokkon** [holgern/beem#72]
#### Expected behavior
The `Signed_Transaction` class should accept existing blockchain transactions, for example to determine the original signer.

#### Actual behavior
`Signed_Transaction` gets along with blockchain transaction from non-appbase nodes, but fails with appbase transactions if those contain a `comment_options` operation with beneficiaries. The constructor fails with `Exception: Unknown CommentOptionExtension`

#### How to reproduce
```python
from beem import Steem
from beem.nodelist import NodeList
from beem.blockchain import Blockchain
from beembase.signedtransactions import Signed_Transaction

s = Steem(node=NodeList().get_nodes(appbase=True, normal=False))
b = Blockchain(steem_instance=s)
for block in b.blocks(start=25304468, stop=25304468):
    for trx in block.transactions:
        try:
            st = Signed_Transaction(trx.copy())
        except Exception:
            print(trx)
            raise
```

Output:
```
{'transaction_id': '4c1d57de6cfeb0966c2e08da85822483c4af37d7', 'ref_block_num': 7551, 'ref_block_prefix': 107790536, 'expiration': '2018-08-23T00:11:06', 'operations': [{'type': 'comment_operation', 'value': {'parent_author': '', 'parent_permlink': 'ifttt', 'author': 'buffalo-goku', 'permlink': 'toincreaseamericasproductivitybanthishttpstcoivgi8aul1n-efyvj7avgx', 'title': "To Increase America's Productivity Ban This... https://t.co/ivgi8aUL1N", 'body': '<blockquote class="twitter-tweet"><p lang="en" dir="ltr">To Increase America&rsquo;s Productivity Ban This&hellip; <a href="https://t.co/ivgi8aUL1N" target="_blank">https://t.co/ivgi8aUL1N</a></p>— Buffalo_Goku🖕 (@F_C_J_R) <a href="https://twitter.com/F_C_J_R/status/914347700178518016?ref_src=twsrc%5Etfw" target="_blank">October 1, 2017</a>\n</blockquote>\n<br />\nfrom Twitter <a href="https://twitter.com/F_C_J_R" target="_blank">https://twitter.com/F_C_J_R</a><br /><br />\nOctober 01, 2017 at 12:34AM<br />\nvia <a href="http://ift.tt/18ufGjY" target="_blank">IFTTT</a><br /><center><hr/><em>Posted from my blog with <a href=\'https://wordpress.org/plugins/steempress/\'>SteemPress</a> : https://nopoliticalcorrectnesshere.com/wp/2017/10/01/to-increase-americas-productivity-ban-this-13/</em><hr/></center>', 'json_metadata': '{"community":"steempress","app":"steempress/1.4","image":[""],"tags":["ifttt","news","nwo","politics","twitter"],"original_link":"https://nopoliticalcorrectnesshere.com/wp/2017/10/01/to-increase-americas-productivity-ban-this-13/"}'}}, {'type': 'comment_options_operation', 'value': {'author': 'buffalo-goku', 'permlink': 'toincreaseamericasproductivitybanthishttpstcoivgi8aul1n-efyvj7avgx', 'max_accepted_payout': {'amount': '1000000000', 'precision': 3, 'nai': '@@000000013'}, 'percent_steem_dollars': 10000, 'allow_votes': True, 'allow_curation_rewards': True, 'extensions': [{'type': 'comment_payout_beneficiaries', 'value': {'beneficiaries': [{'account': 'steempress', 'weight': 1500}]}}]}}, {'type': 'vote_operation', 'value': {'voter': 'buffalo-goku', 'author': 'buffalo-goku', 'permlink': 'toincreaseamericasproductivitybanthishttpstcoivgi8aul1n-efyvj7avgx', 'weight': 10000}}], 'extensions': [], 'signatures': ['203a68c042cddd1b93dd4792893b66f31880c8d410643b46ce43fa4d674012a32b4e670185dcb9c1c5db90ceddb74ecd0046d9bfdf5c0ebde962a6e267f35c6a04'], 'block_num': 25304468, 'transaction_num': 3}
Traceback (most recent call last):
  File "bug_benef.py", line 12, in <module>
    st = Signed_Transaction(trx.copy())
  File "/usr/local/lib/python3.6/site-packages/beembase/signedtransactions.py", line 30, in __init__
    super(Signed_Transaction, self).__init__(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/beemgraphenebase/signedtransactions.py", line 66, in __init__
    kwargs['operations'] = Array([opklass(a) for a in kwargs["operations"]])
  File "/usr/local/lib/python3.6/site-packages/beemgraphenebase/signedtransactions.py", line 66, in <listcomp>
    kwargs['operations'] = Array([opklass(a) for a in kwargs["operations"]])
  File "/usr/local/lib/python3.6/site-packages/beembase/objects.py", line 79, in __init__
    super(Operation, self).__init__(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/beemgraphenebase/objects.py", line 54, in __init__
    self.op = klass(op["value"])
  File "/usr/local/lib/python3.6/site-packages/beembase/operations.py", line 364, in __init__
    extensions = Array([CommentOptionExtensions(o) for o in kwargs["extensions"]])
  File "/usr/local/lib/python3.6/site-packages/beembase/operations.py", line 364, in <listcomp>
    extensions = Array([CommentOptionExtensions(o) for o in kwargs["extensions"]])
  File "/usr/local/lib/python3.6/site-packages/beembase/objects.py", line 268, in __init__
    raise Exception("Unknown CommentOptionExtension")
Exception: Unknown CommentOptionExtension
```
Changing the nodelist to non-appbase (`appbase=False, normal=True`) gives no output as expected. Changing the block number to 25304469, containing a `comment_options` operation without beneficiaries, works for both appbase and non-appbase.

It seems `Signed_Transaction` doesn't get along with the additional appbase `type`/`value` pair in `extensions`:
```python
'extensions': [{
  'type': 'comment_payout_beneficiaries', 
  'value': {
    'beneficiaries': [
      {'account': 'steempress', 'weight': 1500}
    ]
}]
```

#### Environment
```
# python --version
Python 3.6.6
# beempy --version
beempy, version 0.19.54
```

**Comment by crokkon** [comment]
Just realized that `beempy verify` is affected in the same way:
* appbase:
```
$ beempy set nodes https://api.steemit.com
$ beempy verify --trx 3 25304468
[...]
  File "/usr/local/lib/python3.6/site-packages/beembase/objects.py", line 268, in __init__
    raise Exception("Unknown CommentOptionExtension")
Exception: Unknown CommentOptionExtension
```
* non-appbase:
```
$ beempy set nodes wss://rpc.steemviz.com
$ beempy verify --trx 3 25304468
+-----+-------------------------------------------------------+--------------+
| trx | Signer key                                            | Account      |
+-----+-------------------------------------------------------+--------------+
| 3   | STM8UsiAQE3nCAjs719SJ3cBTJzH8KQ5jwAemcrn6DVVCphQvjH2j | buffalo-goku |
+-----+-------------------------------------------------------+--------------+
```

**Comment by holgern** [comment]
The described is fixed after applying #77
